python/ray/data/_internal/execution/operators/actor_pool_map_operator.py

```diff
@@ -503,19 +534,59 @@ def pending_to_running(self, ready_ref: ray.ObjectRef) -> bool:
         self._actor_locations[actor] = ray.get(ready_ref)
         return True
 
+    def running_to_restarting(
+        self, actor: ray.actor.ActorHandle, ready_ref: ray.ObjectRef
+    ) -> bool:
+        """Mark the actor corresponding to the provided ready future as restarting.
+
+        Args:
+            actor: The running actor to add as restarting to the pool.
+            ready_ref: The ready future for the actor that we wish to mark as
+            restarting.
+
+        Returns:
+            Whether the actor was still running. This can return False if the actor had
+            already been killed.
+        """
+        if actor not in self._num_tasks_in_flight:
+            # The actor has been removed from the pool before becoming restarting.
+            return False
+        self._remove_actor(actor)
+        self._restarting_actors[ready_ref] = actor
+        return True
+
+    def restarting_to_running(self, ready_ref: ray.ObjectRef) -> bool:
+        """Mark the actor corresponding to the provided ready future as running, making
+        the actor pickable.
+
+        Args:
+            ready_ref: The ready future for the actor that we wish to mark as running.
+
+        Returns:
+            Whether the actor was still restarting. This can return False if the actor
+            had already been killed.
+        """
+        if ready_ref not in self._restarting_actors:
+            # The actor has been removed from the pool before becoming running.
+            return False
+        actor = self._restarting_actors.pop(ready_ref)
+        self._num_tasks_in_flight[actor] = 0
```

Comment:
We need to keep the old `_num_tasks_in_flights` and restore it here

Reply:
Ah I see, makes sense